File: python/DesignPatternExamples_python/strategy/strategy_entryinformation.py
## @file
#  @brief
#  Implementation of the
#  @ref DesignPatternExamples_python.strategy.strategy_entryinformation.EntryInformation "EntryInformation"
#  class as used in the @ref strategy_pattern.

##  Represents an individual with a Name, Age, and Height.
class EntryInformation:

    ## Constructor.
    #
    #  @param name
    #         Name of the individual.
    #  @param age
    #         Age of the individual (in years).
    #  @param height
    #         Height of the individual (in inches).
    def __init__(self, name : str, age : int, height : int) -> None:
        self.Name = name
        self.Age = age
        self.Height = height

    ## @var Name
    #       Name of this individual.
    #  @var Age
    #       Age of this individual, in years.
    #  @var Height
    #       Height of this individual, in inches.


    ## Convert the entry into a string.  In this case, a formatted
    #  string where name, age, and height are confined to columns.
    #
    #  @returns
    #     A string representation of this instance.
    def ToString(self) -> str:
        return "{0:6} {1:3} {2:3}\"".format(self.Name, self.Age, self.Height)

File: rust/design_pattern_examples_rust/src/strategy/strategy_entryinformation.rs
//! Contains the EntryInformation struct that holds information about each
//! entry that can be sorted.

//-----------------------------------------------------------------------------

use std::fmt::Display;

//-----------------------------------------------------------------------------

/// Represents an individual with a Name, Age, and Height.
#[derive(Clone)]
pub struct EntryInformation {
    /// Name of this individual.
    pub name: String,
    /// Age of this individual, in years.
    pub age: i32,
    /// Height of this individual, in inches.
    pub height: i32,
}

impl EntryInformation {
    /// Constructor
    ///
    /// # Parameters
    /// - name
    ///
    ///   Name of an individual
    /// - age
    ///
    ///   Age of an individual (in years)
    /// - height
    ///
    ///   Height of an individual (in inches)
    ///
    /// # Returns
    /// Returns a new instance of the EntryInformation struct.
    pub fn new(name: &str, age: i32, height: i32) -> EntryInformation {
        EntryInformation {
            name: name.to_string(),
            age,
            height,
        }
    }
}

impl Display for EntryInformation {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        f.write_fmt(format_args!("{0:6} {1:3} {2:3}\"", self.name, self.age, self.height))
    }
}
